src/main/kotlin/in/tazj/k8s/letsencrypt/acme/Route53Responder.kt

```diff
@@ -72,6 +72,17 @@ class Route53Responder(val route53: AmazonRoute53) : DnsResponder {
                 .reduce { acc, zone ->
                     if (zone.name.length > acc.name.length) {
                         zone
+                    } else if (zone.name.length == acc.name.length) {
```

Comment:
The actual name should be equal for this to work, so I guess that is a better comparison.

In AWS the only way to have two zones with the same name is if they are of different types (i.e. private vs. public).

I'll change it~